app/controllers/rooms_controller.rb

```diff
@@ -1,29 +1,46 @@
 class RoomsController < ApplicationController
-  before_action :load_rooms, except: %i(show)
+  before_action :load_rooms, only: %i(index)
   before_action :load_room, only: %i(show)
+  before_action :load_attributes
 
-  def index
-    @attributes = RoomAttribute.pluck(:name, :id)
-  end
+  # routes
+
+  def index; end
 
   def show; end
 
   private
 
+  # before action
+
+  def load_attributes
+    @attributes = RoomAttribute.pluck(:name, :id)
+  end
+
   def load_room
-    @room = Room.find_by id: params[:id]
-    return if @room
+    @room = Room.find_by id: prs[:id]
+    return if @room.any?
 
     flash[:danger] = t :not_found
     redirect_to root_path
   end
 
   def load_rooms
-    @rooms = Room.room_filter room_params
-    flash.now[:danger] = t :empty_result unless @rooms.any?
+    attrs = filter_params[:attributes] || []
+    @rooms = Room.status_is(:available)
+                 .price_sort(filter_params[:sort])
+                 .name_search(filter_params[:keyword])
+                 .pagination_at(filter_params[:page])
+
+    @rooms = @rooms.has_attributes attrs if attrs.length > 1
+    return if @rooms.any?
+
+    flash.now[:danger] = t :empty
   end
 
-  def room_params
-    params.permit(:sort, :keyword, :page, :locale, attributes: [])
+  # params permit
+
+  def filter_params
+    params.permit(:id, :sort, :keyword, :page, :locale, attributes: [])
```

Comment:
`attributes: []` ko có field j ah?

Reply:
Em cũng không biết nữa em tham khảo[ select tag with multiple attribute](https://blog.geckox.mx/2021/05/09/select-tag-multiple-attribute-has-many-association-in-rails.html)
Thì thấy họ sử dụng cú pháp `: []` để lấy một Array trong params thông qua permit

### View
```erb
 <%= f.select :category_ids,
    options_from_collection_for_select(categories, :id, :name),
    { include_blank: false },
    multiple: true,
    class: 'select multiple-select',
    size: 5
  %>
```
### Controller
```ruby
 params.require(:post)
      .permit(:title,
              :content,
              category_ids: [])
```